src/qibo/core/distcircuit.py

```diff
@@ -94,10 +94,9 @@ def _add(self, gate: gates.Gate):
 
         Also checks that there are sufficient qubits to use as global.
         """
-        if K.name not in {"qibotf", "qibojit"}:
-            raise_error(NotImplementedError, "Distributed circuit is implemented "
-                                             "only for the qibotf and qibojit "
-                                             "backends.")
+        if not K.supports_multigpu:
+            raise_error(NotImplementedError, f"Distributed circuit is not supported "
+                                              "by the {K.name} backend.")
```

Comment:
I think we could move this check in the `DistributedCircuit.__init__` (line 49 same file). Currently we have a `sys.platform == "darwin"` check there which I believe we can change to `if K.supports_multigpu`, since the mac os check is now done within the multi-GPU backends to set the `self.supports_multigpu` property.